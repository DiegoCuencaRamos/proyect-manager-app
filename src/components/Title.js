import React, { useContext } from 'react'
import { Link, useParams } from 'react-router-dom'
import { ProjectContext } from '../contexts/ProjectContext'

const Title = ({ title, description, isDashboard }) => {
    // Variables
    const isProyect = useContext(ProjectContext)
    const proyectId = isProyect ? null : useParams().proyectId

    // Render
    return (
        <div className="page-header">
            <div className="container page-header__wrapper">
                <div>
                    <h1 className="page-header__title">{title}</h1>
                    <p className="page-header__text">{description}</p>
                </div>
                {
                    isDashboard && (
                        <Link 
                            to={isProyect ? '/add-proyect' : `/add-task/${proyectId}`}
                            className="button"
                        >Add{isProyect ? 'Proyect' : 'Task'}</Link>
                    )  
                }
            </div>
        </div>
    )
}

export default Title