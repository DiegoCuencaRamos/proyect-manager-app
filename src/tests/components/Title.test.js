import React from 'react'
import { shallow, render } from 'enzyme'
import { MemoryRouter } from 'react-router'
import DashboardContext from '../../contexts/dashboard-context'
import Title from '../../components/Title'

const isProyect = true

test('Should render Title correctly', () => {
    const wrapper = shallow(<Title title="Some title" description="Some description" />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render Title correctly on Dashboard Proyect page', () => {
    const wrapper = render(
        <MemoryRouter>
            <DashboardContext.Provider value={{ isProyect }}>
                <Title
                    title="Some title" 
                    description="Some description"
                    isDashboard={true}
                />
                </DashboardContext.Provider>
        </MemoryRouter>
        )
    expect(wrapper).toMatchSnapshot()
})

test('Should render Title correctly on Dashboard Task page', () => {
    const wrapper = render(
        <MemoryRouter>
            <Title
                title="Some title" 
                description="Some description"
                isDashboard={true}
            />
        </MemoryRouter>
        )
    expect(wrapper).toMatchSnapshot()
})