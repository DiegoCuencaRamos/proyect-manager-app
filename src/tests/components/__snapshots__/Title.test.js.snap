// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Title correctly on Dashboard Proyect page 1`] = `
<div
  class="page-header"
>
  <div
    class="container page-header__wrapper"
  >
    <div>
      <h1
        class="page-header__title"
      >
        Some title
      </h1>
      <p
        class="page-header__text"
      >
        Some description
      </p>
    </div>
    <a
      class="button"
      href="/add-proyect"
    >
      AddProyect
    </a>
  </div>
</div>
`;

exports[`Should render Title correctly on Dashboard Task page 1`] = `
<div
  class="page-header"
>
  <div
    class="container page-header__wrapper"
  >
    <div>
      <h1
        class="page-header__title"
      >
        Some title
      </h1>
      <p
        class="page-header__text"
      >
        Some description
      </p>
    </div>
    <a
      class="button"
      href="/add-task"
    >
      AddTask
    </a>
  </div>
</div>
`;

exports[`Should render Title correctly with no context 1`] = `
<div
  className="page-header"
>
  <div
    className="container page-header__wrapper"
  >
    <div>
      <h1
        className="page-header__title"
      >
        Some title
      </h1>
      <p
        className="page-header__text"
      >
        Some description
      </p>
    </div>
  </div>
</div>
`;
